### DISCUSSION TITLE
TODOs 📝

### DISCUSSION THREAD

**Discussion by rocktimsaikia** [AnimechanOrg/animechan#87] [General]
:bulb: The below things need to be done in the same order also (Important):
- [x] Typescript migration #88 
- [x] Use DrizzleORM and migrate to PlanetScale #90 
- [ ] Release V2 of API where new and more standardized routes will introduced. While still keeping the previous endpoints otherwise it will break a lot of applications using Animechan.
- [ ] Persist rate limits between deployments and restarts by using a remote Redis server
- [ ] Introduce token-based rate limiting. People with unique tokens can bypass the default rate limit
 to an extent.
- [ ] Write tests.

cc @melosh101

**Comment by rocktimsaikia** [comment]
Typescript migration is done #88

**Comment by melosh101** [comment]
Should we migrate the frontend to typescript too?

**Reply by melosh101** [reply]
Nvm I'm blind

**Comment by melosh101** [comment]
Is it important to use mongodb? Since you wanna move to another "orm" than mongoose and typorm's mongodb implementation is kinda bad, in my opinion (it is marked as experimental in the docs), and due to the size of the dataset size and consistent data structure it would be wise to move to a SQL database like mysql or postgresql.

**Reply by rocktimsaikia** [reply]
Is there any free db hosting service for Postgresql or SQLite that you know of or used? I used MongoDB for this project cause Atlas has provided free hosting and connectivity till now.

**Reply by melosh101** [reply]
We could use planetscale, would also allow for us maintainers to have a "dev db" we can mess around with and then push to prod or roll back. 
The free hobby tier includes:
- 5 GB storage included
- 1 billion row reads per month ($1 per additional billion)
- 10 million row writes per month ($1.50 per additional million)

Which should be plenty for our usage. And now speaking of services I think we can keep upstash in mind for when we in the future make the ratelimits persist between server restarts, its a server less redis database service with plenty everhead at their starting teir

**Reply by rocktimsaikia** [reply]
Good suggestions. I will take a look. Most likely will go with Planetscale with Drizzle ORM.

**Reply by rocktimsaikia** [reply]
Drizzle and Planetscale migration is done :heavy_check_mark: #90
